### PR TITLE
Update the Project References section based on recent perf investigations

### PR DESCRIPTION
**Pull request by amcasey**

### CODE REVIEW

**Review by orta — APPROVED** [review]

File: Performance.md [hunk 1]
```diff
@@ -181,16 +186,30 @@ Tests can also be broken into their own project.
...
-As a heuristic, if things are related enough to go in the same folder, they belong in the same project.
-Beyond that, avoid massive or tiny projects.
-If one project is larger than all the others combined, that's a warning sign.
-Similarly, it's best to avoid having dozens of single-file projects because the overhead adds up.
-
 You can [read up more about project references here](https://www.typescriptlang.org/docs/handbook/project-references.html).
 
+## Existing Code
+
+When a workspace becomes so large that it's hard for the editor to handle (and you've used [performance tracing](#providing-performance-traces) to confirm that there are no hotspots, making scale the most likely culprit) , it can be helpful to break it down into a collection of projects that reference each other.
```
Comment (DanielRosenwasser) [inline Performance.md:193]:
```suggestion
When a workspace becomes so large that it's hard for the editor to handle (and you've used [performance tracing](#providing-performance-traces) to confirm that there are no hotspots, making scale the most likely culprit), it can be helpful to break it down into a collection of projects that reference each other.
```

Comment (DanielRosenwasser) [reply Performance.md:193]: How would a user know that scale is the culprit?

File: Performance.md [hunk 2]
```diff
@@ -181,16 +186,30 @@ Tests can also be broken into their own project.
...
+Otherwise the process is more ad hoc - you may be able to follow the directory structure or you may have to use carefully chosen `include` and `exclude` globs.
+Some things to keep in mind:
+* Aim for evenly-sized projects - avoid having a single humongous project with lots of tiny satellites
+* Try to group together files that will be edited together - this will limit the number of projects the editor needs to load
+* Separating out test code can help prevent product code from accidentally depending on it
+
+## Performance Considerations
+
+As with any encapsulation mechanism, projects come with a cost.
+For example, if all projects depend on the same typings (e.g. a popular UI framework), some parts of those typings will be checked repeatedly - once for each project consuming them.
```
Comment (DanielRosenwasser) [inline Performance.md:204]:
```suggestion
For example, if all projects depend on the same packages (e.g. a popular UI framework), some parts of that package's types will be checked repeatedly - once for each project consuming them.
```

File: Performance.md [hunk 3]
```diff
@@ -181,16 +186,30 @@ Tests can also be broken into their own project.
...
+Empirically, it seems that (for a workspace with more than one project) 5-20 projects is an appropriate range - fewer may result in editor slowdowns and more may result in excessive overhead.
+Some good reasons to split out a project:
+* It has a different output location (e.g. because it's a package in a monorepo)
+* It requires different settings (e.g. `lib` or `moduleResolution`)
+* It contains global declarations that you want to scope (either for encapsulation or to limit expensive global rebuilds)
+* The editor's language service runs out of memory when trying to process the code as a single project
+  * In this case, you will want to set `"disableReferencedProjectLoad": true` and `"disableSolutionSearching": true` to limit project loading while editing
```
Comment (DanielRosenwasser) [inline Performance.md:211]:
```suggestion
    * In this case, you will want to set `"disableReferencedProjectLoad": true` and `"disableSolutionSearching": true` to limit project loading while editing
```

**Review by DanielRosenwasser — APPROVED** [review]

File: Performance.md (hunk 1 above)
Comment (amcasey) [reply Performance.md:193]: The lack of hotspots.  That's what I was attempting to convey with the parenthetical, so please feel free to reword it to make that clearer.

File: Performance.md (hunk 3 above)
Comment (amcasey) [reply Performance.md:211]: It was supposed to be a sub-bullet.  Did I get the syntax wrong?

Comment (DanielRosenwasser) [reply Performance.md:211]: It's just a consistency thing, I think we use 4 spaces elsewhere. Not a big deal.

Comment (amcasey) [reply Performance.md:211]: Oops, I read the diff backwards.

File: Performance.md (hunk 1 above)
Comment (amcasey) [reply Performance.md]: I'm going to merge so I don't lose track of this.  Please feel free to reword in a future PR.
